src/model.cpp

```diff
@@ -51,35 +60,42 @@ model::model(const recipe& rec, const domain_decomposition& decomp):
             PL(2);
         });
 
-
     // Create event lane buffers.
     // There is one set for each epoch: current (0) and next (1).
     // For each epoch there is one lane for each cell in the cell group.
-    event_lanes_[0].resize(communicator_.num_local_cells());
-    event_lanes_[1].resize(communicator_.num_local_cells());
+    event_lanes_[0].resize(num_local_cells);
+    event_lanes_[1].resize(num_local_cells);
 }
 
 void model::reset() {
     t_ = 0.;
 
+    // Reset cell group state.
```

Comment:
No irony implied: Thank you for these comments. Reading the code will take x times as long. So these small sentences speed up understanding the code a lot

Reply:
Pleasure!
The list of tasks in the `reset` method was growing, so it was time to add some comments to make the steps clearer.